sdk/python/kfp/compiler/compiler.py

```diff
@@ -189,29 +190,31 @@ def _create_pipeline(
 
         return pipeline_spec
 
-    def _write_pipeline_spec_json(
+    def _write_pipeline_spec_yaml(
         self,
         pipeline_spec: pipeline_spec_pb2.PipelineSpec,
         output_path: str,
     ) -> None:
-        """Writes pipeline spec into a json file.
+        """Writes pipeline spec into a YAML file.
 
         Args:
             pipeline_spec: IR pipeline spec.
             ouput_path: The file path to be written.
 
         Raises:
-            ValueError: if the specified output path doesn't end with the
-                acceptable extention.
+            ValueError: if the specified output path doesn't end with a
+                .yaml extention.
         """
-        json_text = json_format.MessageToJson(pipeline_spec, sort_keys=True)
+        # sort keys=False throughout retains PipelineSpec's order
+        json_dict = json.loads(
```

Comment:
Note: `google.protobuf` does not have a protobuffer to yaml converter. Therefore, the options are to:

1) Write our own
2) Use another library that does this conversion directly (adding a dependency)
3) Convert to JSON, then to YAML (adds no additional dependencies)

I chose option 3, since this is a very cheap operation.

Reply:
Option 3 sounds good to me as well.